### PR TITLE
Add Metric Support

### PR DESCRIPTION
**Pull request by MrAlias**
Add `Export` method to the `Exporter` to implement the `"go.opentelemetry.io/otel/sdk/export/metric".Exporter` interface.

Add transform logic to the `transform` package to handle OpenTelemetry `Record`s and transform them into New Relic `Metric`s.

Update existing end-to-end test validate metric exporting.

### CODE REVIEW

File: newrelic/internal/transform/metric.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+var ErrUnimplementedAgg = errors.New("unimplemented aggregator")
+
+// Record transforms an OpenTelemetry Record into a Metric.
+//
+// An ErrUnimplementedAgg error is returned for unimplemented Aggregators.
+func Record(service string, r metricsdk.Record) (telemetry.Metric, error) {
+	desc := r.Descriptor()
+	attrs := attributes(service, desc, r.Labels())
+	switch a := r.Aggregator().(type) {
+	case aggregator.MinMaxSumCount:
```
Comment (lykkin) [inline newrelic/internal/transform/metric.go:27]: is this a complete enumeration of aggregator types at this point, or is this just a starting point?

Comment (MrAlias) [reply newrelic/internal/transform/metric.go:27]: Is the only ones currently exported. [There are a few other aggregations](https://github.com/open-telemetry/opentelemetry-go/tree/master/sdk/metric/aggregator), but these are the only two used for all (3) current instruments.

File: newrelic/internal/transform/identifiers.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ const (
 	errorCodeAttrKey    = "error.code"
 	errorMessageAttrKey = "error.message"
 
+	serviceNameAttrKey = "service.name"
```
Comment (purple4reina) [inline newrelic/internal/transform/identifiers.go:10]: Just a comment: It's interesting to me that our telemetry sdk adds the `service.name` attr for spans but _not_ for metrics.

File: newrelic/internal/transform/metric.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		return minMaxSumCount(desc, attrs, a)
+	case aggregator.Sum:
+		return sum(desc, attrs, a)
+	}
+	return nil, ErrUnimplementedAgg
```
Comment (purple4reina) [inline newrelic/internal/transform/metric.go:32]: What happens in otel when this error is returned? Do the metrics get retried next harvest period?

File: newrelic/internal/transform/identifiers.go (hunk 2 above)
Comment (MrAlias) [reply newrelic/internal/transform/identifiers.go:10]: Yeah, the `ServiceName` is a first class field in the JSON span structure we send to New Relic, so it became a part of the Span struct in the SDK. The inclusion in the metric is just by knowing it needs to be in the free form attributes. Which makes sense why it is kind of the same way in the SDK. Might be worth refactoring the design in the SDK, but the same could be said of the Metric API I guess. :shrug:

File: newrelic/internal/transform/metric.go (hunk 3 above)
Comment (MrAlias) [reply newrelic/internal/transform/metric.go:32]: That's ultimately going to be up to the configured Controller. This error will get rolled up all the way back to the caller of the `Export` method this exporter implements on the `Exporter` interface. This method [is called by the Controller](https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/metric/controller/push/push.go#L181) in the OpenTelemetry Go SDK. The default "push" Controller (currently the only controller) has a configurable error handler function that [by default prints the error to STDERR](https://github.com/open-telemetry/opentelemetry-go/blob/3008c1bf025b4710f3760c974ec679ae658e0d76/sdk/metric/sdk.go#L326).

**Pull request merged**